tests/test_client.py

```diff
@@ -143,3 +146,84 @@ def test_bad_endpoints(self):
                     "account": "",
                 },
             )
+
+
+@patch("pasqal_cloud.client.Auth0TokenProvider", FakeAuth0AuthenticationSuccess)
+class TestEnvTestSDK(TestSDKCommonAttributes):
+    @pytest.mark.parametrize(
+        "env, core_endpoint_expected",
+        [
+            (None, 'https://apis.pasqal.cloud/core-fast'),
+            ('prod', 'https://apis.pasqal.cloud/core-fast'),
+            ('preprod', 'https://apis.preprod.pasqal.cloud/core-fast'),
+            ('dev', 'https://apis.dev.pasqal.cloud/core-fast')
+        ]
+    )
+    def test_existing_env(self, env, core_endpoint_expected):
+        sdk = SDK(
+            project_id=self.project_id,
+            username=self.username,
+            password=self.password,
+            env=env
+        )
+        assert sdk._client.endpoints.core == core_endpoint_expected
+
+    def test_init_sdk_with_unknown_env(self):
+        with pytest.raises(ValueError, match="This environment does not exist, valid options are"):
+            SDK(
+                project_id=self.project_id,
+                username=self.username,
+                password=self.password,
+                env="wrong"
+            )
+
+    @pytest.mark.filterwarnings(
+        "ignore:The parameters 'endpoints' and 'auth0' are deprecated, from now use 'env' instead"
+    )
+    def test_endpoints_and_auth0_and_env_given(self):
+        with pytest.raises(
+                TypeError,
+                match="The `env` argument is incompatible with `endpoints` and `auth0` arguments."
+        ):
+            auth0 = Auth0Conf(domain="new_domain")
+            SDK(
+                project_id=self.project_id,
+                username=self.username,
+                password=self.password,
+                endpoints=Endpoints(core=self.new_core_endpoint),
+                auth0=auth0,
+                env="prod"
+            )
+
+    @pytest.mark.filterwarnings(
+        "ignore:The parameters 'endpoints' and 'auth0' are deprecated, from now use 'env' instead"
+    )
+    def test_endpoints_and_env_given(self):
+        with pytest.raises(
+                TypeError,
+                match="The `env` argument is incompatible with `endpoints` and `auth0` arguments."
+        ):
+            SDK(
+                project_id=self.project_id,
+                username=self.username,
+                password=self.password,
+                endpoints=Endpoints(core=self.new_core_endpoint),
+                env="prod"
+            )
+
+    @pytest.mark.filterwarnings(
+        "ignore:The parameters 'endpoints' and 'auth0' are deprecated, from now use 'env' instead"
+    )
+    def test_auth0_and_env_given(self):
```

Comment:
Can you merge the 3 test above using pytest.mark.parametrize ?

Reply:
Working properly now :)